float32[] elements
int32 reward
